src/queue.ts

```diff
@@ -0,0 +1,155 @@
+import BN = require('bn.js')
+import gql from 'graphql-tag'
+import { Observable } from 'rxjs'
+import { Arc, IApolloQueryOptions } from './arc'
+import { Address } from './types'
+import { realMathToNumber } from './utils'
+
+export interface IQueue {
+  id: string
+  dao: Address
+  name?: string
+}
+
+export interface IQueueState {
+  id: string
+  // scheme: IScheme
+  paramsHash: string
+  threshold: BN
+  thresholdConst: BN
+  dao: Address
+  votingMachine: Address
+  queuedVoteRequiredPercentage: number
+  queuedVotePeriodLimit: number // in seconds (?)
+  boostedVotePeriodLimit: number
+  preBoostedVotePeriodLimit: number
+  limitExponentValue: number
+  quietEndingPeriod: number // in seconds (?)
+  proposingRepReward: BN // in REP
+  votersReputationLossRatio: number // in 1000's
+  minimumDaoBounty: BN // in GEN
+  daoBountyConst: number // ?
+  activationTime: number
+  voteOnBehalf: Address
+}
+
+export class Queue {
+  public static search(
+    options: {
+      dao?: Address,
+      name?: string
+    },
+    context: Arc,
+    apolloQueryOptions: IApolloQueryOptions = {}
+): Observable<Queue[]> {
+    let where = ''
+    for (const key of Object.keys(options)) {
+      const value = (options as any)[key]
+      if (value !== undefined) {
+        where += `${key}: "${value}"\n`
+      }
+    }
+
+    const query = gql` {
+     gpqueues (where: {${where}}) {
+       id
+       dao { id }
+       scheme {
+         id
+         name
+       }
+     }
+   }`
+    const itemMap = (item: any): Queue => {
+      return new Queue(
+        item.id,
+        item.dao.id,
+        item.scheme.name,
+        context
+      )
+    }
+
+    return context._getObservableList(query, itemMap, apolloQueryOptions) as Observable<Queue[]>
+  }
+  public id: Address
+  public dao: Address
+  public name: string
+
+  constructor(id: Address, dao: Address, name: string, public context: Arc) {
```

Comment:
I think its always nice to have the constructor at the top of the class, and the instance variables too

Reply:
my linter does this automatically, and what it does is:

- first static methods
-then instance variables
- then the constructor
- then the normal methods

for me this is fine :-)